### PR TITLE
Root Cert Download from Main API Page

### PR DESCRIPTION
**Pull request by johnccr**
Hello, I wanted to facilitate Mobile security testing using ZAP by allowing download of the CA certificate directly from the API home page.

Please consider merging my changes and let me know if you have any questions or comments.

Regards,
Juan C Calderon

### CODE REVIEW

**Comment by kingthorin** [comment]
> Run 'gradlew spotlessApply' to fix these violations.

**Comment by johnccr** [comment]
Ran spotlessapply and uploaded the code changes that broke the CI build. Can you retry the build or should I try a different compile option to ensure all checks are performed?

**Comment by kingthorin** [comment]
Don't worry about snyk. Travis passed that's the actual build.

**Comment by johnccr** [comment]
Ok cool, then I think all worked out. Only your reviews are missing. Let me know if you have any doubt or comment on the changed functionality

**Comment by thc202** [comment]
@johnccr could you "allow edits by maintainers" so we can update the branch to target main? (Unless you want to do it yourself.)

**Comment by johnccr** [comment]
> 
> 
> @johnccr could you "allow edits by maintainers" so we can update the branch to target main? (Unless you want to do it yourself.)

Sure, but I don't see the checkbox on the pull request for some reason :(, can you please advice what can I do?

**Comment by thc202** [comment]
Are you checking in the right hand side under the "participants" section? If it's not possible to enable it could you rebase yourself?

**Comment by johnccr** [comment]
> 
> 
> Are you checking in the right hand side under the "participants" section? If it's not possible to enable it could you rebase yourself?

Yes, under participants I only see me, rick and you :( 
image

I would need help with the rebase as well, sorry I am not literate at Github, can you help? you can contact me directly at @johnccr  by Twitter, we can have a quick screen share to do it at once if you have the time to spare.

**Comment by thc202** [comment]
The following git commands should do that:
```
git fetch upstream main
git rebase upstream/main
git push origin -f
```
we can then change the target branch in the pull request to main.

File: zap/src/main/dist/lang/Messages_es_ES.properties [hunk 1]
```diff
@@ -123,6 +123,10 @@ api.home.links.header=<h2>Enlaces</h2>
 api.home.links.api.enabled=<li><a href\="/UI"> API local</a></li>
 api.home.links.api.disabled=<li>API local desactivada - Puede ser activada a trav\u00e9s de Herramientas / Opciones... /</li>
 api.home.links.online=<li><a href\="https\://www.owasp.org/index.php/ZAP">Pagina de ZAP</a></li>\n<li><a href\="https\://github.com/zaproxy/zaproxy/wiki/Introduction">ZAP Wiki</a></li>\n<li><a href\="https\://groups.google.com/group/zaproxy-users">Grupo de Usuarios de ZAP</a></li>\n<li><a href\="https\://groups.google.com/group/zaproxy-develop">Grupo de desarrolladores ZAP</a></li>\n<li><a href\="https\://github.com/zaproxy/zaproxy/issues">Reportar una falla</a></li>
+api.home.cacert=<h2>Evitar Advertencias de HTTPS</h2>\
```
Comment (thc202) [inline zap/src/main/dist/lang/Messages_es_ES.properties:126]: This should be reverted, only the main file should be changed. Translations happen in Crowdin.

File: zap/src/main/java/org/zaproxy/zap/extension/api/CoreAPI.java [hunk 2]
```diff
@@ -1386,7 +1386,8 @@ public HttpMessage handleApiOther(HttpMessage msg, String name, JSONObject param
                     String response = sw.toString();
                     msg.setResponseHeader(
                             API.getDefaultResponseHeader(
-                                    "application/pkix-cert;", response.length()));
+                                            "application/pkix-cert;", response.length())
+                                    + "Content-Disposition: attachment; filename=\"ZapCACert.cer\"\r\n");
```
Comment (thc202) [inline zap/src/main/java/org/zaproxy/zap/extension/api/CoreAPI.java:1390]: ZAP should be all upper case.

**Comment by johnccr** [comment]
Darn I feel useless, the commands you send below do not work on my zap repository and I am lost on how to fix the request.

Can you just drop the pull request and apply the patch on your side? changes are as few that it would take more time to fix the pull request than to simply copy and paste changed lines.

Regards,
Juan Carlos Calderón Rojas
 

    On Wednesday, January 27, 2021, 5:34:40 AM CST, thc202 <notifications@github.com> wrote:  
 
 


The following git commands should do that:
git fetch upstream main
git rebase upstream/main
git push origin -f

we can then change the target branch in the pull request to main.

**Comment by kingthorin** [comment]
I'll take care of re-creating and addressing those last few comments.

**Comment by johnccr** [comment]
I really appreciate it Rick, I hope my next contribution is less painful.

Regards,
Juan Carlos Calderón Rojas
 

    On Thursday, January 28, 2021, 11:14:00 AM CST, Rick M <notifications@github.com> wrote:  
 
 


I'll take care of re-creating and addressing those last few comments.

**Comment by kingthorin** [comment]
See: https://github.com/zaproxy/zaproxy/pull/6428

**Pull request closed without merging**